arangod/Aql/QuerySnippet.cpp

```diff
@@ -668,14 +668,24 @@ auto QuerySnippet::prepareFirstBranch(
           if (localGraphNode->isDisjoint()) {
             if (found->second == server) {
               myExp.emplace(shard);
+            } else {
+              // the target server does not have anything to do with the particular
+              // collection (e.g. because the collection's shards are all on other
+              // servers), but it may be asked for this collection, because vertex
+              // collections are registered _globally_ with the TraversalNode and
+              // not on a per-target server basis.
+              // so in order to serve later lookups for this collection, we insert
+              // an empty string into the collection->shard map.
+              // on lookup, we will react to this.
+              localGraphNode->addCollectionToShard(aqlCollection->name(), "");
```

Comment:
Changing the `_collectionToShard` map to contain `optional<string>`s would make it a little more explicit, but its not necessary.

Reply:
I think this is just left here as a security net, this should never show up anymore.